### DISCUSSION TITLE
Default sorting

### DISCUSSION THREAD

**Discussion by makgr** [yajra/yajrabox.com#25] [Q&A]
![sorting table](https://github.com/yajra/yajrabox.com/assets/19414854/58dd566f-55d8-44fc-b625-6ead52922cbd)

Default sorting is not working in laravel. How to fix this??

**Comment by yajra** [comment]
You can define default orders via JS or the builder

```php
->orderBy(1, 'desc')
->orderBy(2, 'asc')

->order([[1, 'desc'], [2, 'asc']])
```

**Reply by makgr** [reply]
yes. But not working.
![order](https://github.com/yajra/yajrabox.com/assets/19414854/66676a37-943a-4d5f-817a-436ec8f3ba48)

**Reply by yajra** [reply]
Can you provide your server-side code? Make sure you don't have any hard-coded orderBy clause in your query.

**Reply by makgr** [reply]
Serverside code
![serverside](https://github.com/yajra/yajrabox.com/assets/19414854/b0c6bb8d-52a1-4c15-b96b-8605f4a48509)

**Reply by yajra** [reply]
Sorting is disabled on [added column](https://yajrabox.com/docs/laravel-datatables/10.0/add-column#main-content). Make sure you have configured your column to something like:

```js
columns:[
  {data: 'date', name: 'created_at'},
  ...
]
```

**Reply by makgr** [reply]
Ok. Then how to do it without addColumn if I need the addColumn feature?

**Reply by yajra** [reply]
You can add a column via SQL.

**Reply by makgr** [reply]
If i need this type of conditional column then how to do it.
![condition](https://github.com/yajra/yajrabox.com/assets/19414854/f64495c6-0329-4e84-89d7-430a48b61899)

**Reply by yajra** [reply]
Using MySQL, something like:

```php
->addSelectRaw("if(notification_status = 1, 'Seen', 'Unread') as status")
```
